File: ScriptGraph/TemplateScriptNode/ComponentGetEnable/CGGetEnable.js
/**
 * @file CGGetEnable.js
 * @author liujiacheng
 * @date 2021/8/19
 * @brief CGGetEnable.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGGetEnable extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput(index) {
    let v1 = this.inputs[0]();
    if (v1.isInstanceOf('Component')) {
      return v1.enabled;
    } else {
      return null;
    }
  }
}

exports.CGGetEnable = CGGetEnable;

File: ScriptGraph/TemplateScriptNode/ComponentGetEnable/CGGetEnable.lua
--
-- Created by IntelliJ IDEA.
-- User: xuyuan
-- Date: 2021/7/28
-- Time: 10:39
-- To change this template use File | Settings | File Templates.
--

local CGGetEnable = CGGetEnable or {}
CGGetEnable.__index = CGGetEnable

function CGGetEnable.new()
    local self = setmetatable({}, CGGetEnable)
    self.inputs = {}
    self.outputs = {}
    return self
end

function CGGetEnable:setInput(index, func)
    self.inputs[index] = func
end

function CGGetEnable:getOutput(index)
    if self.inputs[0] then
        if self.inputs[0]():isInstanceOf("Component") then
            return  self.inputs[0]().enabled
        end
    end
    return false
end

return CGGetEnable
